src/app/clusters/water-heater-management-server/water-heater-management-server.h

```diff
@@ -0,0 +1,132 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/MessageDef/StatusIB.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/CHIPError.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WaterHeaterManagement {
+
+class Delegate
+{
+public:
+    Delegate()          = default;
+    virtual ~Delegate() = default;
+
+    void SetEndpointId(EndpointId aEndpoint) { mEndpointId = aEndpoint; }
+
+    /**
+     * @brief Delegate should implement a handler to start boosting the water temperature as required.
+     *        Upon receipt, the Water Heater SHALL transition into the BOOST state, which SHALL cause the water in the
+     *        tank (or the TargetPercentage of the water, if included) to be heated towards the set point (or the
+     *        TemporarySetpoint, if included), which in turn may cause a call for heat, even if the mode is OFF, or
+     *        is TIMED and it is during one of the Off periods.
+     *
+     * @param duration  Indicates the time period in seconds for which the BOOST state is activated before it automatically reverts
```

Comment:
Nit - the alignment across all fields could be neater

Reply:
This seems to be style used elsewhere